File: src/main/java/com/support_App/model/Technician.java
package com.support_App.model;


import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;

import java.util.List;

@Entity
public class Technician extends User {
    @OneToMany(mappedBy = "technician", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    @JsonIgnore
    private List<SupportTicket> supportTickets;
}

File: IT_FrontEnd/src/app/model/Technician/technician.ts
import {User} from "../User/user";
import {Role} from "../../enums/role";
import {SupportTicket} from "../SupportTicket/support-ticket";

export class Technician extends User{

  supportTickets: SupportTicket[] = [];
  constructor() {
    super();
    this.role = Role.Technician

  }
}
